ast-nodes/statement.dart

```diff
@@ -1,9 +1,36 @@
+import '../syntax-error.dart';
 import 'node.dart';
 import '../lexer.dart';
+import 'while-loop.dart';
+import 'for-loop.dart';
+import 'if-statement.dart';
+import 'routine-call.dart';
+import 'assignment.dart';
 
 /// An abstract statement.
 abstract class Statement implements Node {
```

Comment:
Yeah, btw, this is where we deviate from the official grammar. The current class structure has `SimpleDeclaration` to be a descendant of `Statement`, which means that `Statement.parse` should also recognize and delegate `VariableDeclaration` and `TypeDeclaration`. Sawwy

Reply:
Wait, isn't it recognized in `Declaration`?

Reply:
In our structure, `Declaration` is a statement